commands/create.go

```diff
@@ -0,0 +1,190 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/gsclientgen"
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/gsctl/config"
+	"github.com/giantswarm/gsctl/util"
+)
+
+var (
+	// CreateCommand is the command to create things
+	CreateCommand = &cobra.Command{
+		Use:   "create",
+		Short: "Create things, like kubectl configuration, or key-pairs",
+		Long:  `Lets you create things like key-pairs`,
+	}
+
+	// CreateKeypairCommand performs the "create keypair" function
+	CreateKeypairCommand = &cobra.Command{
+		Use:     "keypair",
+		Short:   "Create key-pair",
+		Long:    `Creates a new key-pair for a cluster`,
+		PreRunE: checkAddKeypair,
+		Run:     addKeypair,
+	}
+
+	// CreateKubeconfigCommand performs the "create kubeconfig" function
+	CreateKubeconfigCommand = &cobra.Command{
+		Use:     "kubeconfig",
+		Short:   "Configure kubectl",
+		Long:    `Modifies kubectl configuration to access your Giant Swarm Kubernetes cluster`,
+		PreRunE: checkCreateKubeconfig,
+		Run:     createKubeconfig,
+	}
+)
+
+func init() {
+	CreateKeypairCommand.Flags().StringVarP(&cmdClusterID, "cluster", "c", "", "ID of the cluster to create a key-pair for")
+	CreateKeypairCommand.Flags().StringVarP(&cmdDescription, "description", "d", "", "Description for the key-pair")
+
+	CreateKubeconfigCommand.Flags().StringVarP(&cmdClusterID, "cluster", "c", "", "ID of the cluster")
+	CreateKubeconfigCommand.Flags().StringVarP(&cmdDescription, "description", "d", "", "Description for the key-pair")
+	CreateKubeconfigCommand.Flags().IntVarP(&cmdTTLDays, "ttl", "", 30, "Duration until expiry of the created key-pair in days")
+
+	// subcommands
+	CreateCommand.AddCommand(CreateKeypairCommand, CreateKubeconfigCommand)
+}
+
+func checkAddKeypair(cmd *cobra.Command, args []string) error {
+	if config.Config.Token == "" {
+		return errors.New("You are not logged in. Use '" + config.ProgramName + " login' to log in.")
+	}
+	if cmdClusterID == "" {
+		// use default cluster if possible
+		clusterID, _ := config.GetDefaultCluster()
+		if clusterID != "" {
+			cmdClusterID = clusterID
+		} else {
+			return errors.New("No cluster given. Please use the -c/--cluster flag to set a cluster ID.")
+		}
+	}
+	if cmdDescription == "" {
+		return errors.New("No description given. Please use the -d/--description flag to set a description.")
+	}
+	return nil
+}
+
+func addKeypair(cmd *cobra.Command, args []string) {
+	client := gsclientgen.NewDefaultApi()
+	authHeader := "giantswarm " + config.Config.Token
+	// TODO: Make ttl configurable
+	ttlHours := int32(3)
+	addKeyPairBody := gsclientgen.AddKeyPairBody{Description: cmdDescription, TtlHours: ttlHours}
+	keypairResponse, _, err := client.AddKeyPair(authHeader, cmdClusterID, addKeyPairBody)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if keypairResponse.StatusCode == 10000 {
```

Comment:
This comes probably from https://github.com/giantswarm/api-schema/blob/master/api_schema.go#L4. It feels a little bit odd that client specific status codes need to be handles in the command line tool. This separation makes no sense to me design wise. Anyway, instead of the magic number, you can use the apischema library.